cpp/ps0004/problem 4 p301F Salwa معضلة الاسكي فاليو.cpp

```diff
@@ -4,20 +4,31 @@ using namespace std;
 
 int main()
 {
-cout<<"--------------------------------------------" <<endl;
-cout<<"Problem 0003"<< endl;
-
-    int num1;
+    // prompting user for number 1/1.5
+    float num;
     cout << "enter a number of ASCII value \n";
-    cin>> num1;
+    cin >> num;
+    int temp = num; // to compare with user input
 
-    if(num1>=32 &num1<=126 )
-{
-   cout <<"the character for  "<< num1 << " is "<<char (num1) << "\n";
-}
-else
-{
-    cout <<"Invalid character/s or zero value \n";
-}
+    // checking invalidaty of user's input 2/4
+	// printing the number and the opposite printable character 2/2
+    if(num == 0)
+    {
+        cout << "Invalid character/s or zero value \n";
+    }
+    else if(num != temp)
+    {
+        cout << "Invalid: a floating point number! \n";
+    }
+    else if(num >= 32 && num <= 126)
+    {
+       cout << "the character for  " << num << " is " << char (num) << "\n";
+    }
+    else
+    {
+        cout << "Invalid: number is out of range! \n";
+    }
+    
     return 0;
 }
+// good work 👍 6/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [ ] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more